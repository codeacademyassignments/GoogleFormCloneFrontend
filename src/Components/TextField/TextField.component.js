import React, { Component } from 'react';
import { TextInput } from 'react-native';
import PropTypes from 'prop-types';
import style from './TextField.style';

export default class TextField extends Component {
  render() {
    const { index, onChange } = this.props;
    return (
      <TextInput style={style.text} onChangeText={(text) => { onChange(index, text); }} />
    );
  }
}

TextField.propTypes = {
  index: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};
