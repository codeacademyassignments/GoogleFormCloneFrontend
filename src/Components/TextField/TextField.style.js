import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  text: {
    backgroundColor: 'red',
  },
});
