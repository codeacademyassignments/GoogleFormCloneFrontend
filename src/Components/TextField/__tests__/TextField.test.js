import React from 'react';
import renderer from 'react-test-renderer';
import TextField from '../TextField.component';

describe('TextField', () => {
  it('should match with old snapshot', () => {
    const tree = renderer.create(<TextField index={0} onChange={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
