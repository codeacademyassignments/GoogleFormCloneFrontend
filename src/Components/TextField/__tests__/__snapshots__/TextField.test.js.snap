// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should match with old snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
  underlineColorAndroid="transparent"
/>
`;
